metaseq/trainer.py

```diff
@@ -398,13 +398,13 @@ def save_checkpoint(
                 def perform_save():
                     try:
                         logger.info(f"Beginning asynchronous torch.save to {filename}")
-                        if async_callback_fn is not None:
-                            async_callback_fn(filename)
+                        async_callback_fn(filename)
```

Comment:
Should we gate this behind save_to_NFS logic somewhere?

Reply:
Similar to answer on Stephen's comment, I am currently passing async_callback as None if  `save_to_NFS == False`